ml-agents-envs/mlagents_envs/tests/test_rpc_utils.py

```diff
@@ -85,6 +85,7 @@ def generate_compressed_data(in_array: np.ndarray) -> bytes:
 def generate_compressed_proto_obs(in_array: np.ndarray) -> ObservationProto:
     obs_proto = ObservationProto()
     obs_proto.compressed_data = generate_compressed_data(in_array)
+    obs_proto.mapping.extend(list(range(len(in_array))))
```

Comment:
This needs some more unit tests. I think you should cover
* 1 channel, no mapping (old case)
* 3 channels, no mapping (old case)
* >3 channels, no mapping (old case)
* 1 channel, mapping
* 3 channel, mapping
* >3 channels with a non-trivial mapping (e.g. 4 channels and stacking=2)

Reply:
Sure I'll add those.
But are we allowing compressed observation with 1 channel? I thought it should always be uncompressed.

Reply:
> compressed observation with 1 channel

Grayscale visual observations, and those can be stacked.